lib/src/base-styles/_nx-forms.scss

```diff
@@ -14,6 +14,12 @@
     color: var(--nx-swatch-red-40);
   }
 
+  &__required-field-notice {
```

Comment:
We're failing to specify the top margin here, and getting the browser default for `<p>` tags.  I think a better approach would be to ensure (in the `RequiredFieldNotice` component) that the element includes both this class and the `nx-p` class (e.g, use `<NxP>` instead of `<p>`) so that we just get the normal `nx-p` margins and don't have to worry about margins at all in _this_ class.